src/patterns.jl

```diff
@@ -14,6 +14,14 @@ AbstractPattern
 """
 abstract type AbstractPattern end
 
+"""
+    isshared(pattern)
+
+Indicates whether patterns share memory (mutate).
```

Comment:
Not the same concept. Our non-shared patterns are still mutating when we perform operations like `union!`

Reply:
Is the following more clear?

> Indicates whether patterns share memory, allowing operators to mutate their arguments.

> Our non-shared patterns are still mutating when we perform operations like union!

Internal allocations and mutations of said new allocations aren't usually what Julia developers refer to as `mutating functions`.

Reply:
See https://github.com/adrhill/SparseConnectivityTracer.jl/commit/c5e4f51af8c710804cfc63d36fbdcb176772d4ae